frontstage/templates/sign-in/sign-in.html

```diff
@@ -9,39 +9,33 @@
 
 {% block main %}
 
-    {% if errorType %}
-        {% set errorData = [] %}
+{% if errorType %}
+        {% set errorData = {} %}
+        {% set errorTitle = 'There is a problem with your answer' %}
         {% if errorType == 'failed' %}
-            {% set errorTitle = 'There is 1 error on this page' %}
-            {% do errorData.append(
-                {
-                    "text": "Incorrect email or password",
+            {% set errorData = {
+                    "text": "Sign in details are invalid. Enter an email address and a password",
                     "url": "#username_error",
                     "classes": "ons-js-inpagelink"
-                }
-            ) %}
-        {% elif errorType|length > 1 %}
-            {% set errorTitle = 'There are ' ~ errorType|length ~ ' errors on this page' %}
-            {% for error in errorType %}
-                {% do errorData.append(
-                    {
-                        "text": errorType[error][0],
-                        "url": "#username_error",
-                        "classes": "ons-js-inpagelink"
-                    }
-                ) %}
-            {% endfor %}
-        {% else %}
-            {% set errorTitle = 'There is 1 error on this page' %}
-            {% for error in errorType %}
-                {% do errorData.append(
-                    {
-                        "text": errorType[error][0],
-                        "url": "#username_error",
-                        "classes": "ons-js-inpagelink"
-                    }
-                ) %}
-            {% endfor %}
+                } %}
+        {% elif errorType['username'] and errorType['password'] %}
```

Comment:
This is odd duplication with line 33, if the dict has {'type': {'username': ['Email Address is required'] regardless of the password being there it will set the same errorData